pkg/models/projectconfigs.go

```diff
@@ -27,7 +27,8 @@ type DeviceMetricsConfig struct {
 }
 
 type ExposedMetric struct {
-	Name       string   `json:"name" yaml:"name"`
-	Labels     []string `json:"labels" yaml:"labels"`
-	Properties []string `json:"properties" yaml:"properties"`
+	Name            string   `json:"name" yaml:"name"`
+	Labels          []string `json:"labels" yaml:"labels"`
+	Properties      []string `json:"properties" yaml:"properties"`
+	WhitelistedTags []string `json:"whitelistedTags" yaml:"whitelistedTags"`
```

Comment:
This would be a better data format

```
 	WhitelistedTags []WhitelistedTag `json:"whitelistedTags" yaml:"whitelistedTags"` 
}

type WhitelistedTag struct {
  Key string
  Values []string
}
```

Reply:
Sounds reasonable. Conveniently, the format imples that key-only vs key+value pairs are mutually exclusive, which makes sense.